composition/composition_handler.go

```diff
@@ -40,8 +40,15 @@ func (agg *CompositionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	for _, res := range results {
 		if res.Err == nil && res.Content != nil {
 
+			if res.Content.HttpStatusCode() == 301 || res.Content.HttpStatusCode() == 302 || res.Content.HttpStatusCode() == 303 {
+				copyHeaders(res.Content.HttpHeader(), w.Header(), ForwardResponseHeaders)
+				w.WriteHeader(res.Content.HttpStatusCode())
+				return
+			}
+
 			if res.Content.Reader() != nil {
 				copyHeaders(res.Content.HttpHeader(), w.Header(), ForwardResponseHeaders)
+				w.WriteHeader(res.Content.HttpStatusCode())
```

Comment:
This will always flush the header for the first successful Content. Wouldn't this block error statuscodes from being passed to the caller?